pkg/awsclient/ec2_helper.go

```diff
@@ -0,0 +1,188 @@
+package awsclient
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+)
+
+// FIXME: Add tags
+
+// EnsureCIDRAccess ensures that for the given load balancer, the specified CIDR
+// blocks, and only those blocks may access it.
+// cidrBlocks always goes from 6443/TCP to 6443/TCP and is IPv4 only
+// TODO: Expand to IPv6. This could be done by regular expression
+func (c *awsClient) EnsureCIDRAccess(loadBalancerName, securityGroupName, vpcId string, cidrBlocks []string) error {
+	// first need to see if the SecurityGroup exists, and if it does not, create it and populate its ingressCIDR permissions
+	// If the SecurityGroup DOES exist, then make sure it only has the permissions we are receiving here.
+	securityGroup, err := c.findSecurityGroupByName(securityGroupName)
+	if err != nil {
+		return err
+	}
+	if securityGroup == nil {
+		// group does not exist, create it
+		securityGroup, err = c.createSecurityGroup(securityGroupName, vpcId)
+		if err != nil {
+			return err
+		}
+	}
+	// At this point, securityGroup is unified, no matter how we got it:
+	// finding it, or creating it and so now we can reconcile the rules
+
+	var rulesToRemove, rulesToAdd []*ec2.IpPermission
+
+	// When processing all this SecurityGroup's ingress rules we compare
+	// to cidrBlocks, but that doesn't always hit the expected set, so
+	// this is a map to see if we have done just that. Any which are
+	// false were not processed.
+	seenExpectedRules := make(map[string]bool)
+	// init map
+	for _, cidrBlock := range cidrBlocks {
+		seenExpectedRules[cidrBlock] = false
+	}
+	// For each ingress rule for the security group,
+Outer:
```

Comment:
That's how `go fmt` decided to format it 🤷‍♀ 

I agree that it looks visually strange, but that's what came out.

The purpose of the `Outer` label is to  skip processing the rule once we've encountered it in the inner loop (lines 52-60) because by that time, it's already been processed.